src/publish-registry.ts

```diff
@@ -30,46 +30,52 @@ export default async function publishRegistry(dt: FS, allPackages: AllPackages,
     const [log, logResult] = logger();
     log("=== Publishing types-registry ===");
 
-    const { version: oldVersion, highestSemverVersion, contentHash: oldContentHash, lastModified } =
-        await fetchAndProcessNpmInfo(packageName, client);
+    const { npmVersion, highestSemverVersion, npmContentHash, lastModified } =
+        await fetchAndProcessNpmInfo(typesRegistry, client);
+    assert.strictEqual(npmVersion.major, 0);
+    assert.strictEqual(npmVersion.minor, 1);
 
     // Don't include not-needed packages in the registry.
     const registryJsonData = await withNpmCache(client, cachedClient => generateRegistry(allPackages.allLatestTypings(), cachedClient));
     const registry = JSON.stringify(registryJsonData);
     const newContentHash = computeHash(registry);
+    const newVersion = `0.1.${npmVersion.patch + 1}`;
 
-    assert.strictEqual(oldVersion.major, 0);
-    assert.strictEqual(oldVersion.minor, 1);
-    const newVersion = `0.1.${oldVersion.patch + 1}`;
-    const packageJson = generatePackageJson(newVersion, newContentHash);
-    await generate(registry, packageJson);
-
-    const publishClient = () => NpmPublishClient.create({ defaultTag: "next" });
-    if (!highestSemverVersion.equals(oldVersion)) {
-        // There was an error in the last publish and types-registry wasn't validated.
-        // This may have just been due to a timeout, so test if types-registry@next is a subset of the one we're about to publish.
-        // If so, we should just update it to "latest" now.
-        log("Old version of types-registry was never tagged latest, so updating");
-        await validateIsSubset(await readNotNeededPackages(dt), log);
-        await (await publishClient()).tag(packageName, highestSemverVersion.versionString, "latest");
-    } else if (oldContentHash !== newContentHash && isAWeekAfter(lastModified)) {
-        log("New packages have been added, so publishing a new registry.");
-        await publish(await publishClient(), packageJson, newVersion, dry, log);
-    } else {
-        const reason = oldContentHash === newContentHash ? "Was modified less than a week ago" : "No new packages published";
-        log(`${reason}, so no need to publish new registry.`);
-        // Just making sure...
-        await validate(log);
+    await publishToRegistry("github");
+    await publishToRegistry("npm");
+    await writeLog("publish-registry.md", logResult());
+
+    async function publishToRegistry(registryName: "github" | "npm") {
+        const packageName = registryName === "github" ? "@definitelytyped/" + typesRegistry : typesRegistry;

```

Comment:
this line, plus the calls to publishToRegistry, are almost all the new code except for registryName in generateJson. Everything else is renames or reordering.